### PR TITLE
Add DeleteRow method

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/12594)

### CODE REVIEW

File: components/script/dom/htmltableelement.rs [hunk 1]
```diff
@@ -338,6 +348,22 @@ impl HTMLTableElementMethods for HTMLTableElement {
...
+        // Step 1.
+        if index == -1 {
+            index = rows.len() as i32 - 1;
+        }
+        // Step 2.
+        if index < 0 || index as usize >= rows.len() {
+            return Err(Error::IndexSize);
+        }
+        // Step 3.
+        rows.sections.remove(index as usize);
```
Comment (GuillaumeGomez) [inline components/script/dom/htmltableelement.rs]: I'm not sure it does remove the row as expected. If someone could confirm please?

File: components/script/dom/htmltableelement.rs [hunk 2]
```diff
@@ -338,6 +348,22 @@ impl HTMLTableElementMethods for HTMLTableElement {
         Ok(new_row)
     }
 
+    // https://html.spec.whatwg.org/multipage/tables.html#dom-table-deleterow
```
Comment (KiChjang) [inline components/script/dom/htmltableelement.rs]: Tidy will fail on this line.

Comment (GuillaumeGomez) [inline components/script/dom/htmltableelement.rs]: Why?

File: components/script/dom/htmltableelement.rs (hunk 1 above)
Comment (jdm) [inline components/script/dom/htmltableelement.rs]: Write a test?

**Comment by jdm** [comment]
How does `tests/wpt/web-platform-tests/html/semantics/tabular-data/the-tbody-element/deleteRow.html` fare with these changes?

**Comment by jdm** [comment]
Or maybe `tests/wpt/web-platform-tests/dom/nodes/getElementsByClassName-21.htm` instead.

**Comment by KiChjang** [comment]
```
Ran 6788 tests finished in 805.0 seconds.
  • 6787 ran as expected. 1998 tests skipped.
  • 1 tests had unexpected subtest results

Tests with unexpected results:
  ▶ Unexpected subtest result in /html/dom/interfaces.html:
  └ PASS [expected FAIL] HTMLTableElement interface: operation deleteRow(long)

  ▶ Unexpected subtest result in /html/dom/interfaces.html:
  └ PASS [expected FAIL] HTMLTableElement interface: document.createElement("table") must inherit property "deleteRow" with the proper type (13)

  ▶ Unexpected subtest result in /html/dom/interfaces.html:
  └ PASS [expected FAIL] HTMLTableElement interface: calling deleteRow(long) on document.createElement("table") with too few arguments must throw TypeError
```

Looks like no real WPT tests have been fixed by this patch...

**Comment by Ms2ger** [comment]
This doesn't actually remove the element from the dom tree, does it?

**Comment by GuillaumeGomez** [comment]
@Ms2ger: It should now. I'm testing if it does it correctly.

**Comment by GuillaumeGomez** [comment]
I added tests now as well. I think it's ready.

File: tests/wpt/web-platform-tests/dom/nodes/remove-row.html [hunk 3]
```diff
@@ -0,0 +1,50 @@
+<!doctype html>
+<meta charset="utf-8">
+<title></title>
```
Comment (emilio) [inline tests/wpt/web-platform-tests/dom/nodes/remove-row.html]: nit: Add a title here

**Comment by jdm** [comment]
Travis says the MANIFEST.json changes aren't quite right; fix it with `./mach update-manifest`.

**Comment by GuillaumeGomez** [comment]
Updated.

File: tests/wpt/metadata/MANIFEST.json [hunk 4]
```diff
@@ -15226,6 +15226,10 @@
         "url": "/dom/nodes/prepend-on-Document.html"
       },
       {
+        "path": "dom/nodes/remove-row.html",
```
Comment (Ms2ger) [inline tests/wpt/metadata/MANIFEST.json]: This test should go in its proper place: `tests/wpt/web-platform-tests/html/semantics/tabular-data/the-table-element/`.

**Comment by Ms2ger** [comment]
@bors-servo retry p=0
* #12580

**Pull request merged**